src/read.rs

```diff
@@ -645,6 +645,32 @@ impl<R: Read + Seek> ZipArchive<R> {
         Ok(shared)
     }
 
+    /// Returns key and salt
+    pub fn get_aes_key_and_salt(
+        &mut self,
+        file_number: usize,
+    ) -> ZipResult<Option<(AesMode, Vec<u8>, Vec<u8>)>> {
+        let (_, data) = self
+            .shared
+            .files
+            .get_index(file_number)
+            .ok_or(ZipError::FileNotFound)?;
+
+        if !data.encrypted {
+            return Err(ZipError::UnsupportedArchive(ZipError::PASSWORD_REQUIRED));
```

Comment:
Does this mean it isn't encrypted, or that it is and we don't know the key?

Reply:
This means the file is not encrypted. 
I made this clearer with a dedicated error message.